src/module/data/item/ability.mjs

```diff
@@ -182,4 +209,53 @@ export default class AbilityModel extends BaseItemModel {
 
     context.characteristics = Object.entries(ds.CONFIG.characteristics).map(([value, {label}]) => ({value, label}));
   }
+
+  modifyRollData(rollData) {
+    super.modifyRollData(rollData);
+
+    if (this.actor) {
+      rollData.chr = this.actor.system.characteristics[this.powerRoll.characteristic]?.value;
+    }
+  }
+
+  /**
+   * Use an ability, generating a chat message and potentially making a power roll
+   * @returns {Promise<DrawSteelChatMessage>}
+   */
+  async use() {
+    const messageData = {
+      speaker: DrawSteelChatMessage.getSpeaker({actor: this.actor}),
+      type: "abilityUse",
+      rolls: [],
+      content: this.parent.name,
+      system: {
+        uuid: this.parent.uuid
+      }
+    };
+    // TODO: Put the spend in flavor text (e.g. "Spends 5 Essence" or whatever)
+
+    DrawSteelChatMessage.applyRollMode(messageData, "roll");
+
+    if (this.powerRoll.enabled) {
+      const formula = this.powerRoll.formula ? `2d10 + ${this.powerRoll.formula}` : "2d10";
+      const rollData = this.parent.getRollData();
+      const rollOptions = {
+        type: "ability"
+      }; // TODO: Add in Banes & Edges
+      const powerRoll = new PowerRoll(formula, rollData, rollOptions);
```

Comment:
We should consider putting most of this API in the roll class where it can be given parameters and be separate from items and actors. That's for the future though.